xbmc/dbwrappers/mysqldataset.cpp

```diff
@@ -158,9 +168,6 @@ int MysqlDatabase::connect(bool create_new) {
       }
       else if (create_new)
       {
-        char sqlcmd[512];
-        int ret;
```

Comment:
I don't think `ret` needs to be reset, as it won't be used in the second comparison until it's been assigned a new value from the call to `query_with_reconnect()`.